src/utils.ts

```diff
@@ -132,3 +132,41 @@ export const isDefined = (value: any): boolean => value !== void 0;
 
 export const isMatchingPath = (path: string, criteria: string | RegExp) =>
     isRegex(criteria) ? !!path.match(criteria as RegExp) : checkGlob(path, criteria as string);
+
+export const isExpanded = (path: string, state: JsonViewerState): boolean => state.expanded[path];
+
+export const isCollapsed = (path: string, state: JsonViewerState): boolean => !state.expanded[path];
+
+export const isDeepestVisible = (node: HTMLElement): boolean =>
+    ![...node.children].includes(<HTMLElement>node.querySelector('[part=object]'));
+
+export const isAncestor = (currentPath: string, compare: string): boolean => currentPath.startsWith(compare);
+
+export const isExpandedAndNonPrimitive = (path: string, shadowRoot: ShadowRoot): boolean => {
+    const currentFocus = <HTMLElement>shadowRoot.querySelector(`[data-path="${path}"]`);
+    const nextObject = <HTMLElement>currentFocus.querySelector('[part=object]');
+    return [...currentFocus.children].includes(nextObject);
+};
+
+export const isEmptyObject = (currentFocus: HTMLElement) => {
+    const nestedItterable = <HTMLElement>currentFocus.querySelector('[part=object]');
```

Comment:
sorry, `Itterable -> Iterable`

Reply:
No need to be sorry, I'd like to be correct, even about insignificant things like spelling! 

Changes applied